miio/yeelight.py

```diff
@@ -85,6 +85,109 @@ def name(self) -> str:
         """Return the internal name of the bulb."""
         return self.data["name"]
 
+    @property
+    def color_flowing(self) -> bool:
+        """Return whether the color flowing is active."""
+        return bool(int(self.data["flowing"]))
+
+    @property
+    def color_flow_params(self) -> Optional[str]:
+        """Return color flowing params."""
+        if self.color_flowing:
+            return self.data["flow_params"]
+        return None
+
+    @property
+    def delay_off(self) -> int:
+        """Return delay in minute before bulb is off."""
+        return self.data["delayoff"]
+
+    @property
+    def music_mode(self) -> Optional[bool]:
+        """Return whether the music mode is active."""
+        music_on = self.data["music_on"]
+        if music_on:
+            return bool(int(music_on))
+        return None
+
+    @property
+    def moonlight_mode(self) -> Optional[bool]:
+        """Return whether the moonlight mode is active."""
+        active_mode = self.data["active_mode"]
+        if active_mode:
+            return bool(int(active_mode))
+        return None
+
+    @property
+    def moonlight_mode_brightness(self) -> Optional[int]:
+        """Return current moonlight brightness."""
+        nl_br = self.data["nl_br"]
+        if nl_br:
+            return int(self.data["nl_br"])
+        return None
+
+    @property
+    def is_bg_on(self) -> Optional[bool]:
+        """Return whether the background light is on or off."""
+        bg_power = self.data["bg_power"]
+        if bg_power:
+            return bg_power == "on"
+        return None
```

Comment:
Just an idea that may not be relevant depending on how the query works on lights with no background light: it might make sense to have the mainlight as it is, and provide a `lights()` getter that will return a list of all lights (main, background, other?) presented as objects of a `YeelightLight` class. This would remove the need of code duplication (just using a prefixed variable name for background light).

I'm thinking about how one would integrate this downstream, and having a list of exposed lights and using a common API would make it much simpler to handle there (e.g., in homeassistant, creating multiple light instances based on available lights). Just some food for thought for now though, that might need some experiment to get a feel how it would work.

Reply:
Yes it looks like good idea. But "yeelight" for some reason use different names for  color mode property. This is "color_mode" for main lamp and "bg_lmode" for background. But I think it can be hardcoded only for this situation. I'll looked to this.

Reply:
@rytilahti this is done in f43c652. But have problem with cmd output. Could you provide any suggestion how I can show my list of objects?

Reply:
additional commit 612ec6f with fix command output. Output looks like 
```
Name: 
Developer mode: True
Update default on change: True
Delay in minute before off: 0
Music mode: False
Lights: 
[
   Sub Light - Main 
      Power: False
      Brightness: 100
      Color mode: 2
      RGB: None
      HSV: None
      Temperature: 6500
      Color flowing mode: False
      Color flowing parameters: None
]
Moonlight
   Is in mode: None
   Moonlight mode brightness: None
```
or 
```
Name: 
Developer mode: True
Update default on change: True
Delay in minute before off: 0
Music mode: None
Lights: 
[
   Sub Light - Main 
      Power: False
      Brightness: 100
      Color mode: 2
      RGB: None
      HSV: None
      Temperature: 6500
      Color flowing mode: False
      Color flowing parameters: None
, 
   Sub Light - Background 
      Power: False
      Brightness: 100
      Color mode: 1
      RGB: (255, 77, 106)
      HSV: None
      Temperature: None
      Color flowing mode: False
      Color flowing parameters: None
]
Moonlight
   Is in mode: False
   Moonlight mode brightness: 0
```
But I don't know is it norm to use  `__repr__` for this or not.

Reply:
I think using the repr is the simplest way to make it print the list of objects, like you already found out :-) I just tested the PR with `yeelink.light.color1` and the status query is working just fine with all the extra properties.

Reply:
It must be better after 6d2ce92. @rytilahti Is it ok for now?

Reply:
Now output more compact (show only not None params) Look 43ca9b3 
```
Name: 
Update default on change: True
Delay in minute before off: 0
Music mode: False
Developer mode: True
Main light
   Power: False
   Brightness: 100
   Color mode: 2
   Temperature: 6500
   Color flowing mode: False
```
```
Name: 
Update default on change: True
Delay in minute before off: 0
Developer mode: True
Main light
   Power: False
   Brightness: 100
   Color mode: 2
   Temperature: 6500
   Color flowing mode: False
Background light
   Power: False
   Brightness: 100
   Color mode: 1
   RGB: (255, 77, 106)
   Color flowing mode: False
Moonlight
   Is in mode: False
   Moonlight mode brightness: 0
```